hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/AbstractSerializationService.java

```diff
@@ -100,15 +101,17 @@
         this.nullSerializerAdapter = createSerializerAdapter(new ConstantSerializers.NullSerializer());
     }
 
+    // used by jet
     protected AbstractSerializationService(AbstractSerializationService prototype) {
         this.version = prototype.version;
         this.inputOutputFactory = prototype.inputOutputFactory;
         this.classLoader = prototype.classLoader;
         this.managedContext = prototype.managedContext;
         this.globalPartitioningStrategy = prototype.globalPartitioningStrategy;
         this.outputBufferSize = prototype.outputBufferSize;
-        this.bufferPoolThreadLocal = prototype.bufferPoolThreadLocal;
         this.nullSerializerAdapter = prototype.nullSerializerAdapter;
+        this.bufferPoolThreadLocal = new BufferPoolThreadLocal(this, new BufferPoolFactoryImpl(),
+                HazelcastInstanceNotActiveException::new);
```

Comment:
If you expect to use this constructor only from member side, then `HazelcastInstanceNotActiveException::new` is fine. However if this constructor is to be used also by Jet client-side, then you should rather use whichever `notActiveExceptionSupplier` was used to construct the `prototype` service.

Reply:
from what i see `HazelcastInstanceNotActiveException` is used for instance [here](https://github.com/hazelcast/hazelcast/blob/bab4abfb88f26dc8ecba742b0a2c6997672ad0e1/hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/AbstractSerializationService.java#L198) as well, there is no differentiation between client and member, @sancar what are your thoughts on this?

Reply:
@vbekiaris is right. We don't expect `HazelcastInstanceNotActiveException` on the client side. The code that you shared should also be fixed in my opinion.